networkx/algorithms/traversal/tests/test_beamsearch.py

```diff
@@ -25,3 +25,8 @@ def test_wide(self):
         G = nx.cycle_graph(4)
         edges = nx.bfs_beam_edges(G, 0, identity, width=2)
         assert list(edges) == [(0, 3), (0, 1), (3, 2)]
+
+    def test_width_none(self):
+        G = nx.cycle_graph(4)
+        edges = nx.bfs_beam_edges(G, 0, identity, width=None)
+        assert list(edges) == [(0, 3), (0, 1), (3, 2)]
```

Comment:
This is indeed the answer that the function gives for this case, but I'm a little confused as to why the edge (1, 2) isn't included as well? Isn't that edge also included in the component reachable from the source node? I suppose bfs terminates as soon as all of the nodes in the component have reached, so the edge traversal is not guaranteed to give you all of the edges... Does that sound right?

Reply:
Hi @rossbar thank you so much for review:) I think the edge (1,2) isn't included because the child (2) of parent(1) has been visit by another "better" parent(3) before, the child(2) will not be visit by parent(1) again. BFS terminates when all the nodes have been visited. Please let me know if it doesn't make sense to you:)